Nos/Models/FlagOption.swift

```diff
@@ -8,62 +8,82 @@ struct FlagOption: Identifiable, Equatable {
     let description: String?
 
-    /// An optional message that will be displayed when the user has selected a particular flag.
-    let info: String?
+    /// An optional closure that returns a message to display when the user selects a particular flag.
+    /// The closure takes the title of the previously selected flag as an optional `String` 
+    /// and returns an optional message to provide additional context or details for the current flag.
+    /// - Parameter String?: The title of the previously selected flag, if available.
+    /// - Returns: A message related to the current flag based on the previous selection, 
+    /// or `nil` if no message is provided.
+    let info: ((String?) -> String?)?
 
     /// A unique identifier for the flagging option, based on the `title`.
     var id: String { title }
 
     /// The specific category thet the selected flagging option falls in.
-    var category: FlagCategory
+    let category: FlagCategory
 
-    /// `FlagOption` instances representing different categories of content that can be flagged.
-    static let flagContentCategories: [FlagOption] = [
-        FlagOption(
-            title: String(localized: .localizable.flagContentSpamTitle),
-            description: nil,
-            info: nil,
-            category: .report(ReportCategoryType.spam)
-        ),
+    /// Creates a list of `FlagOption` categories based on the provided flag target.
+    /// - Parameter flagTarget: The target of the report.
+    /// - Returns: An array of `FlagOption` categories,  modified based on the `flagTarget`.
+    static func createFlagCategories(for flagTarget: ReportTarget) -> [FlagOption] {
+        if case .author = flagTarget {
+            return [spam, harassment, nsfw, illegal, other]
+        } else {
+            return [spam, harassment, nsfw, illegal, impersonation, other]
+        }
+    }
+
+    /// `FlagOption` instances representing different categories of how a content can can be flagged.
+    static let flagContentSendOptions: [FlagOption] = [
         FlagOption(
-            title: String(localized: .localizable.flagContentHarassmentTitle),
-            description: String(localized: .localizable.flagContentHarassmentDescription),
-            info: nil,
-            category: .report(ReportCategoryType.harassment)
+            title: String(localized: .localizable.flagSendToNosTitle),
+            description: String(localized: .localizable.flagSendToNosDescription),
+            info: { selectedTitle in
+                guard let selectedTitle = selectedTitle else { return nil }
+                return String(localized: .localizable.flagUserSendToNosInfo(selectedTitle))
+            },
+            category: .privacy(.sendToNos)
         ),
         FlagOption(
-            title: "NSFW",
-            description: String(localized: .localizable.flagContentNudityDescription),
+            title: String(localized: .localizable.flagPubiclyTitle),
```

Comment:
Just caught a typo here -- I know the change is in Localizable.xcstrings but this is where I saw it.

```suggestion
            title: String(localized: .localizable.flagPubliclyTitle),
```

Same below for the `flagPubliclyDescription`.

Reply:
🕵️